### ISSUE TITLE
Update user docs to more accurately describe mtree_set operation

### ISSUE THREAD

**Issue by maxgillett** [maticnetwork/miden#400]
The documentation for `mtree_set` currently states that the Merkle tree with the old root is removed from the advice set map in the advice provider, but not that the new tree resulting from the update is also inserted into the advice set. This should be clarified to prevent confusion.

**Comment by Dominik1999** [comment]
So to clarify: At the moment, the docs state: 

```
Updates a node in the Merkle tree with root R at depth d and index i to value V. R' is the Merkle root of the resulting tree. Merkle tree with root R must be present in the advice provider, otherwise execution fails. At the end of the operation Merkle tree with root R is removed from the advice provider.
```

My questions: 
- Is in this case `advice provider` equal to `Stack`?
- Is it sufficient to change the last sentence of the above description to `At the end of the operation Merkle tree with root R is removed from and the new Merkle tree with root R' is inserted to the advice provider.`

**Comment by Dominik1999** [comment]
Got it, thanks

**Comment by grjte** [comment]
Closed by #476
